src/js/common/schemaform/SelectiveArrayField.jsx

```diff
@@ -0,0 +1,203 @@
+import _ from 'lodash/fp';
+import classNames from 'classnames';
+import Scroll from 'react-scroll';
+
+
+import {
+  toIdSchema,
+  getDefaultFormState,
+  deepEquals
+} from 'react-jsonschema-form/lib/utils';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+
+class SelectiveArrayField extends React.Component {
+  // This fills in an empty item in the array if it has minItems set
+  // so that schema validation runs against the fields in the first item
+  // in the array. This shouldn't be necessary, but there's a fix in rjsf
+  // that has not been released yet
+  componentDidMount = () => {
+    const { schema, formData = [], registry } = this.props;
+    if (schema.minItems > 0 && formData.length === 0) {
+      this.props.onChange(Array(schema.minItems).fill(
+        getDefaultFormState(schema.additionalItems, undefined, registry.definitions)
+      ));
+    }
+  }
+
+  shouldComponentUpdate = (nextProps, nextState) => {
+    return !deepEquals(this.props, nextProps) || nextState !== this.state;
+  }
+
+  onItemChange = (indexToChange, value) => {
+    const newItems = _.set(indexToChange, value, this.props.formData || []);
+    this.props.onChange(newItems);
+  }
+
+  getItemSchema = (index, selectedFields) => {
+    const schema = this.props.schema;
+    const itemSchema = schema.items.length > index
+      ? schema.items[index]
+      : schema.additionalItems;
+
+    const properties = _.pick(selectedFields, itemSchema.properties);
+    const required = _.intersection(selectedFields, itemSchema.required);
+
+    return _.assign(itemSchema, {
+      properties,
+      required
+    });
+  }
+
+  scrollToTop = () => {
+    setTimeout(() => {
+      scroller.scrollTo(`topOfTable_${this.props.idSchema.$id}`, {
+        duration: 500,
+        delay: 0,
+        smooth: true,
+        offset: -60
+      });
+    }, 100);
+  }
+
+  scrollToRow = (id) => {
+    setTimeout(() => {
+      scroller.scrollTo(`table_${id}`, {
+        duration: 500,
+        delay: 0,
+        smooth: true,
+        offset: 0
+      });
+    }, 100);
+  }
+
+  render() {
+    const {
+      uiSchema,
+      errorSchema,
+      idSchema,
+      formData,
+      disabled,
+      readonly,
+      registry,
+      formContext,
+      onBlur,
+      schema
+    } = this.props;
+    const definitions = registry.definitions;
+    const { TitleField, SchemaField } = registry.fields;
+
+    const title = uiSchema['ui:title'] || schema.title;
+    const hideTitle = !!_.get(['ui:options', 'hideTitle'], uiSchema);
+    const description = uiSchema['ui:description'];
+    const textDescription = typeof description === 'string' ? description : null;
+    const DescriptionField = typeof description === 'function'
+      ? uiSchema['ui:description']
+      : null;
+    const hasTitleOrDescription = (!!title && !hideTitle) || !!description;
+    const selectedFields = uiSchema['ui:selectedFields'] || Object.keys(schema.properties);
```

Comment:
I think the alternative is `_.pick()`ing them in the `formConfig`, but we'd also need to make sure we do an `_.intersection()` manually with the fields we want and the fields in `ui:order` we may already have. Rather than doing that in the form config, I did it in here...

...er...I was supposed to, anyhow. Woops. Fixed now!
```js
if (uiSchema.items && uiSchema.items['ui:order']) {
      uiSchema.items['ui:order'] = _.intersection(
        // Make sure ALL the selectedFields are in there
        _.concat(uiSchema.items['ui:order'], selectedFields),
        selectedFields
      );
    }
```

Anywho, I'm interested to see what you're thinking, but the only way I think we can get around this is by doing the work manually in each `formConfig`, which seems silly.